share/rcm.sh.in

```diff
@@ -143,6 +143,12 @@ run_hooks() {
         find "$hook_file" -type f \( \( -user $LOGNAME -perm -100 \) -o -perm -001 \) \
           | sort | while read file; do
             sh -c 'cd -- "`dirname $1`" && ./"`basename $1`"' arg0 "$file"
+            hook_ret=$?
+
+            if [ "$hook_ret" -ne 0 ]; then
+              echo "$when-$direction hook exited non-zero ($hook_ret)" >&2
+              exit $hook_ret # NB. this only exits the while-read; caller must still check and exit
```

Comment:
Whoa, that's good to know. Normally I don't love code comments but I definitely would have missed that quirk.